Gems/Atom/Feature/Common/Code/Source/RayTracing/RayTracingFeatureProcessor.cpp

```diff
@@ -115,11 +118,37 @@ namespace AZ
                         ->IndexBuffer(subMesh.m_indexBufferView)
                 ;
 
-                // create the BLAS object
-                subMesh.m_blas = AZ::RHI::RayTracingBlas::CreateRHIRayTracingBlas();
+                // search for an existing BLAS object for this model
+                RayTracingBlasMap::iterator itBlas = m_blasMap.find(assetId);
```

Comment:
Seems like it would be worthwhile to add a mutex here and in RemoveMesh so that SetMesh/RemoveMesh can be called from multiple threads.

In practice, in O3DE everything is happening on the main thread (at least in the editor) in response to asset system events like OnAssetReady or component sytem events like duplicating, adding, or removing a component. But I don't imagine that will always be the case.